### ISSUE TITLE
Error while receiving message by worker in celery 4.0rc3 on python 2.7.5

### ISSUE THREAD

**Issue by standeck** [celery/celery#3435]
It works fine on python 2.7.12 (Ubuntu 16.04), but not on python 2.7.5 (CentOS 7).
It seems like this issue: [#1637](https://github.com/celery/celery/issues/1637)

system: Linux-3.10.0-327.28.3.el7.x86_64-x86_64-with-centos-7.2.1511-Core
python: 2.7.5
celery: v4.0.0rc3 (0today8)

```
[2016-09-08 11:25:41,578: CRITICAL/MainProcess] Unrecoverable error: TypeError('Struct() argument 1 must be string, not unicode',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/celery/worker/__init__.py", line 201, in start
    self.blueprint.start(self)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 125, in start
    step.start(parent)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 376, in start
    return self.obj.start()
  File "/usr/lib/python2.7/site-packages/celery/worker/consumer/consumer.py", line 315, in start
    blueprint.start(self)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 125, in start
    step.start(parent)
  File "/usr/lib/python2.7/site-packages/celery/worker/consumer/consumer.py", line 582, in start
    c.loop(*c.loop_args())
  File "/usr/lib/python2.7/site-packages/celery/worker/loops.py", line 90, in asynloop
    next(loop)
  File "/usr/lib/python2.7/site-packages/kombu/async/hub.py", line 347, in create_loop
    cb(*cbargs)
  File "/usr/lib/python2.7/site-packages/celery/concurrency/asynpool.py", line 288, in on_result_readable
    next(it)
```

**Comment by AndyMan1** [comment]
Piggybacking, I'm getting almost the exact same error, same file, different line. 
Linux: 3.10.0-123.6.3.el7.x86_64
python: 2.7.5
celery: 4.0.0rc3
```
[2016-09-09 00:41:14,764: CRITICAL/MainProcess] Unrecoverable error: TypeError('Struct() argument 1 must be string, not unicode',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/celery/worker/__init__.py", line 201, in start
    self.blueprint.start(self)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 125, in start
    step.start(parent)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 376, in start
    return self.obj.start()
  File "/usr/lib/python2.7/site-packages/celery/worker/consumer/consumer.py", line 315, in start
    blueprint.start(self)
  File "/usr/lib/python2.7/site-packages/celery/bootsteps.py", line 125, in start
    step.start(parent)
  File "/usr/lib/python2.7/site-packages/celery/worker/consumer/consumer.py", line 582, in start
    c.loop(*c.loop_args())
  File "/usr/lib/python2.7/site-packages/celery/worker/loops.py", line 90, in asynloop
    next(loop)
  File "/usr/lib/python2.7/site-packages/kombu/async/hub.py", line 343, in create_loop
    cb(*cbargs)
  File "/usr/lib/python2.7/site-packages/celery/concurrency/asynpool.py", line 288, in on_result_readable
    next(it)
  File "/usr/lib/python2.7/site-packages/celery/concurrency/asynpool.py", line 242, in _recv_message
    body_size, = unpack_from('>i', bufv)
  File "/usr/lib/python2.7/site-packages/celery/concurrency/asynpool.py", line 61, in unpack_from
    return _unpack_from(fmt, view.tobytes())  # <- memoryview
TypeError: Struct() argument 1 must be string, not unicode
```

**Comment by ask** [comment]
Both of these issues should be fixed on 4.0rc5 on pip now.
